### PR TITLE
images: Add CentOS 10

### PR DESCRIPTION
**Pull request by martinpitt**
Drop the "stream" suffix. It's pointless (there is no other variant) and just bloats the name, and we already dropped it from "centos-9-bootc".

Adjust the setup script to skip currently unavailable packages (kpatch, vdo, and wireguard). Install mock from Fedora 40 (the version RHEL/CentOS 10 was branched from), as there is no EPEL 10 yet.

Add initial _manual testmap entries for the project where we want to enable it first.

----

 * [x] image-refresh centos-10
 * [x] image-refresh centos-8-stream

### CODE REVIEW

**Comment by martinpitt** [comment]
I ran `TEST_OS=centos-10 make check` in starter-kit locally and it passed. :heavy_check_mark: So I'm going to enable that next after https://github.com/cockpit-project/starter-kit/pull/824 and landing this.

**Comment by cockpituous** [comment]
image-refresh centos-10 done: https://github.com/cockpit-project/bots/commits/image-refresh-centos-10-20240415-105833

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-centos-10-d8b685c1-20240415-104422/log.html

**Comment by martinpitt** [comment]
I'll build c8s as a regression check.

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-centos-8-stream-a028f377-20240415-110653/log.html

**Review by allisonkarlitskaya — COMMENTED** [review]
Current c10s does come with a default

*doesn't

File: image-trigger [hunk 1]
```diff
@@ -41,6 +41,7 @@ REFRESH = {
     "centos-8-stream": {},
     "centos-9-bootc": {},
     "centos-9-stream": {},
+    "centos-10": {},
```
Comment (allisonkarlitskaya) [inline image-trigger:44]: We're going to have to be careful about this if we start using the name "centos-10" in tests.  iirc, in testing farm tests we paste "-stream" back onto the end of the OS name if it's CentOS...

File: images/scripts/rhel.setup [hunk 2]
```diff
@@ -304,13 +310,33 @@ if [ "$DO_BUILD" -eq 1 ]; then
     # install mock from EPEL
     major=$(cut -d '-' -f 2 <<< "$IMAGE")
     minor=$(cut -d '-' -f 3 <<< "$IMAGE")
-    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-${major}.noarch.rpm
-    dnf install -y rpm-build mock
-    dnf config-manager --set-disabled epel
+    dnf install -y rpm-build
+
+    if [ "${IMAGE#rhel-10*}" != "$IMAGE" ] || [ "${IMAGE#centos-10}" != "$IMAGE" ]; then
+        # RHEL 10 has no EPEL yet, install from Fedora 40
```
Comment (allisonkarlitskaya) [inline images/scripts/rhel.setup:316]: Ugh....

*looks the other way*

File: images/centos-10 [hunk 3]
```diff
@@ -0,0 +1 @@
+centos-10-745153bad16955b473f507a46fa3ba985160a0ec296b31dea5ed2a4cc77ba50c.qcow2
```
Comment (allisonkarlitskaya) [inline images/centos-10:1]: hihi pleased to meet you.

**Review by allisonkarlitskaya — APPROVED** [review]
I think this is ready to go.  See one concern below about handling of the OS name...

File: image-trigger (hunk 1 above)
Comment (martinpitt) [reply image-trigger:44]: Right, I'll adjust that in the PRs where I enable c10s. In our tests we mostly already do `.startswith('centos-9')` and such.

File: images/scripts/rhel.setup (hunk 2 above)
Comment (martinpitt) [reply images/scripts/rhel.setup:316]: We did the same when introducing RHEL 9 back then, see commit 6d09ffa29253c6ec33a5ea1f9b644ce7f5b759bb

File: image-trigger (hunk 1 above)
Comment (martinpitt) [reply image-trigger:44]: See e.g. https://github.com/cockpit-project/starter-kit/pull/839

**Pull request merged**